tooling/backend_interface/src/lib.rs

```diff
@@ -104,6 +107,18 @@ impl Backend {
     fn crs_directory(&self) -> PathBuf {
         self.backend_directory().join("crs")
     }
+
+    fn assert_correct_version(&self) {
+        let binary_path = self.binary_path();
+        if binary_path.to_string_lossy().contains(BACKEND_BARRETENBERG_SEARCH_STR) {
```

Comment:
Why not do a strict check for `acvm-backend-barretenberg`? We can't give any guarantees around custom backends so it's odd to do so if the user has given them a name which includes "barretenberg".

Reply:
This will warn about most `barretenberg` backends. Also, this sis mostly about default backend, a temporary hack which we'll need to solve some other way. Considering, it will WARN user about possibility of version mismatch.